pennylane/transforms/dynamic_one_shot.py

```diff
@@ -104,67 +136,27 @@ def func(x, y):
                 f"Native mid-circuit measurement mode does not support {type(m).__name__} "
                 "measurements."
             )
-    _ = kwargs.get("device", None)
 
     if not tape.shots:
         raise QuantumFunctionError("dynamic_one_shot is only supported with finite shots.")
 
-    samples_present = any(isinstance(mp, SampleMP) for mp in tape.measurements)
-    postselect_present = any(op.postselect is not None for op in tape.operations if is_mcm(op))
-    if postselect_present and samples_present and tape.batch_size is not None:
-        raise ValueError(
-            "Returning qml.sample is not supported when postselecting mid-circuit "
-            "measurements with broadcasting"
-        )
+    aux_tapes = [init_auxiliary_tape(tape)]
 
-    if (batch_size := tape.batch_size) is not None:
-        tapes, broadcast_fn = qml.transforms.broadcast_expand(tape)
-    else:
-        tapes = [tape]
-        broadcast_fn = None
```

Comment:
Moved into `_expand_fn` so that we don't have to manually handle composing the transforms.